tests/test_report.py

```diff
@@ -190,60 +162,230 @@ def test_report_invalid_version(
     db_session.add(scan)
     db_session.commit()
 
-    body = ReportPackageBody(
-        name="c",
-        version="2.0.0",
-        recipient=None,
-        inspector_url=None,
-        additional_information=None,
-    )
-
     with pytest.raises(HTTPException) as e:
-        report_package(body, db_session, graph_client, auth, pypi_client)
+        _lookup_package("c", "2.0.0", db_session)
     assert e.value.status_code == 404
 
 
-def test_report_missing_inspector_url(
-    db_session: Session, auth: AuthenticationData, pypi_client: PyPIServices, graph_client: GraphClient
-):
-    scan = Scan(
-        name="c",
-        version="1.0.0",
-        status=Status.FINISHED,
-        score=0,
-        inspector_url=None,
-        rules=[],
-        download_urls=[],
-        queued_at=datetime.now() - timedelta(seconds=60),
-        queued_by="remmy",
-        pending_at=datetime.now() - timedelta(seconds=30),
-        pending_by="remmy",
-        finished_at=datetime.now() - timedelta(seconds=10),
-        finished_by="remmy",
-        reported_at=None,
-        reported_by=None,
-        fail_reason=None,
-        commit_hash="test commit hash",
-    )
-    db_session.add(scan)
-    db_session.commit()
+def test_report_missing_inspector_url():
+    with pytest.raises(HTTPException) as e:
+        _validate_inspector_url("a", "1.0.0", None, None)
+    assert e.value.status_code == 400
 
-    body = ReportPackageBody(
-        name="c",
-        version="1.0.0",
-        recipient=None,
-        inspector_url=None,
-        additional_information=None,
-    )
 
+@pytest.mark.parametrize(
+    ("body_url", "scan_url"),
+    [
+        ("test url", None),
+        (None, "test url"),
+    ],
+)
+def test_report_inspector_url(body_url: Optional[str], scan_url: Optional[str]):
+    assert "test url" == _validate_inspector_url("a", "1.0.0", body_url, scan_url)
+
+
+@pytest.mark.parametrize(
+    ("body", "scan"),
```

Comment:
nit: not sure if I like these really long decorators. probably fine to leave it but would be nicer to put it in a fixture

Reply:
It's annoying to put fixtures in decorators, so I just didn't bother. Maybe I will revisit it at some point...